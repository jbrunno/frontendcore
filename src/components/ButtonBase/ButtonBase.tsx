import React from 'react';
import {
  ButtonBase as MuiButtonBase,
  ButtonBaseProps as MuiButtonBaseProps,
} from '@mui/material';

export type ButtonBaseProps = MuiButtonBaseProps;

export function ButtonBase(props: ButtonBaseProps) {
  return <MuiButtonBase {...props} />;
}

export default ButtonBase;
