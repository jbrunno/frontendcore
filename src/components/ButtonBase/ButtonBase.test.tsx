import React from 'react';
import { render } from '@test-utils/component-testing';
import { ButtonBase } from './ButtonBase';

test('renders the ButtonBase', () => {
  const { container } = render(<ButtonBase />);
  expect(container).toMatchSnapshot();
});
