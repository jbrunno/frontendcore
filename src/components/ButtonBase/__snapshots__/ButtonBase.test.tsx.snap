// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the ButtonBase 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.emotion-0::-moz-focus-inner {
  border-style: none;
}

.emotion-0.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-0 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-1 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

<div>
  <button
    class="MuiButtonBase-root emotion-0"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiTouchRipple-root emotion-1"
    />
  </button>
</div>
`;
